### PR TITLE
Replace deprecated `log` calls

### PR DESCRIPTION
**Pull request by annando**
Several `log` calls have been replaced. To reduce the amount of files this PR will have a following one with the rest.

### CODE REVIEW

File: mod/pubsub.php [hunk 1]
```diff
@@ -107,8 +107,8 @@ function pubsub_post(App $a)
 {
 	$xml = Network::postdata();
 
-	Logger::log('Feed arrived from ' . $_SERVER['REMOTE_ADDR'] . ' for ' .  DI::args()->getCommand() . ' with user-agent: ' . $_SERVER['HTTP_USER_AGENT']);
-	Logger::log('Data: ' . $xml, Logger::DATA);
+	Logger::notice('Feed arrived from ' . $_SERVER['REMOTE_ADDR'] . ' for ' .  DI::args()->getCommand() . ' with user-agent: ' . $_SERVER['HTTP_USER_AGENT']);
```
Comment (MrPetovan) [inline mod/pubsub.php]:
```suggestion
	Logger::info('Feed arrived from ' . $_SERVER['REMOTE_ADDR'] . ' for ' .  DI::args()->getCommand() . ' with user-agent: ' . $_SERVER['HTTP_USER_AGENT']);
```

File: mod/pubsub.php [hunk 2]
```diff
@@ -149,7 +149,7 @@ function pubsub_post(App $a)
 		hub_post_return();
 	}
 
-	Logger::log('Import item for ' . $nick . ' from ' . $contact['nick'] . ' (' . $contact['id'] . ')');
+	Logger::notice('Import item for ' . $nick . ' from ' . $contact['nick'] . ' (' . $contact['id'] . ')');
```
Comment (MrPetovan) [inline mod/pubsub.php]:
```suggestion
	Logger::info('Import item for ' . $nick . ' from ' . $contact['nick'] . ' (' . $contact['id'] . ')');
```

File: mod/pubsubhubbub.php [hunk 3]
```diff
@@ -60,11 +60,11 @@ function pubsubhubbub_init(App $a) {
 		} elseif ($hub_mode === 'unsubscribe') {
 			$subscribe = 0;
 		} else {
-			Logger::log("Invalid hub_mode=$hub_mode, ignoring.");
+			Logger::notice("Invalid hub_mode=$hub_mode, ignoring.");
 			throw new \Friendica\Network\HTTPException\NotFoundException();
 		}
 
-		Logger::log("$hub_mode request from " . $_SERVER['REMOTE_ADDR']);
+		Logger::notice("$hub_mode request from " . $_SERVER['REMOTE_ADDR']);
```
Comment (MrPetovan) [inline mod/pubsubhubbub.php]:
```suggestion
		Logger::info("$hub_mode request from " . $_SERVER['REMOTE_ADDR']);
```

File: src/Model/Contact.php [hunk 4]
```diff
@@ -2545,7 +2545,7 @@ public static function createFromProbeForUser(int $uid, $url, $network = '')
 				}
 			} elseif ($protocol == Protocol::DIASPORA) {
 				$ret = Diaspora::sendShare($owner, $contact);
-				Logger::log('share returns: ' . $ret);
+				Logger::notice('share returns: ' . $ret);
```
Comment (MrPetovan) [inline src/Model/Contact.php:2562]:
```suggestion
				Logger::info('share returns: ' . $ret);
```

File: src/Model/Contact.php [hunk 5]
```diff
@@ -2554,7 +2554,7 @@ public static function createFromProbeForUser(int $uid, $url, $network = '')
 				}
 
 				$ret = ActivityPub\Transmitter::sendActivity('Follow', $contact['url'], $uid, $activity_id);
-				Logger::log('Follow returns: ' . $ret);
+				Logger::notice('Follow returns: ' . $ret);
```
Comment (MrPetovan) [inline src/Model/Contact.php:2571]:
```suggestion
				Logger::info('Follow returns: ' . $ret);
```

File: src/Model/Contact.php [hunk 6]
```diff
@@ -2791,7 +2791,7 @@ public static function updateBirthdays()
 		$contacts = DBA::select('contact', ['id', 'uid', 'name', 'url', 'bd'], $condition);
 
 		while ($contact = DBA::fetch($contacts)) {
-			Logger::log('update_contact_birthday: ' . $contact['bd']);
+			Logger::notice('update_contact_birthday: ' . $contact['bd']);
```
Comment (MrPetovan) [inline src/Model/Contact.php:2809]:
```suggestion
			Logger::info('update_contact_birthday: ' . $contact['bd']);
```

File: src/Model/Item.php [hunk 7]
```diff
@@ -2305,7 +2305,7 @@ public static function expire(int $uid, int $days, string $network = "", bool $f
 			++$expired;
 		}
 		DBA::close($items);
-		Logger::log('User ' . $uid . ": expired $expired items; expire items: $expire_items, expire notes: $expire_notes, expire starred: $expire_starred, expire photos: $expire_photos");
+		Logger::notice('User ' . $uid . ": expired $expired items; expire items: $expire_items, expire notes: $expire_notes, expire starred: $expire_starred, expire photos: $expire_photos");
```
Comment (MrPetovan) [inline src/Model/Item.php:2308]:
```suggestion
		Logger::info('User ' . $uid . ": expired $expired items; expire items: $expire_items, expire notes: $expire_notes, expire starred: $expire_starred, expire photos: $expire_photos");
```

File: src/Protocol/DFRN.php [hunk 8]
```diff
@@ -1295,7 +1295,7 @@ private static function transformActivity($xpath, $activity, $element)
 	 */
 	private static function processMail($xpath, $mail, $importer)
 	{
-		Logger::log("Processing mails");
+		Logger::notice("Processing mails");
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php:1298]:
```suggestion
		Logger::info("Processing mails");
```

File: src/Protocol/DFRN.php [hunk 9]
```diff
@@ -1397,7 +1397,7 @@ private static function addSuggestion(int $uid, int $cid, int $from_cid, string
 	 */
 	private static function processRelocation($xpath, $relocation, $importer)
 	{
-		Logger::log("Processing relocations");
+		Logger::notice("Processing relocations");
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php:1403]:
```suggestion
		Logger::info("Processing relocations");
```

File: src/Protocol/DFRN.php [hunk 10]
```diff
@@ -1444,7 +1444,7 @@ private static function processRelocation($xpath, $relocation, $importer)
 
 		Contact::updateAvatar($importer["id"], $relocate["avatar"], true);
 
-		Logger::log('Contacts are updated.');
+		Logger::notice('Contacts are updated.');
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php:1450]:
```suggestion
		Logger::info('Contacts are updated.');
```

File: src/Protocol/DFRN.php [hunk 11]
```diff
@@ -1505,7 +1505,7 @@ private static function getEntryType($importer, $item)
 			if ($importer["page-flags"] == User::PAGE_FLAGS_COMMUNITY || $importer["page-flags"] == User::PAGE_FLAGS_PRVGROUP) {
 				$sql_extra = "";
 				$community = true;
-				Logger::log("possible community action");
+				Logger::notice("possible community action");
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php:1511]:
```suggestion
				Logger::info("possible community action");
```

File: src/Protocol/DFRN.php [hunk 12]
```diff
@@ -1525,7 +1525,7 @@ private static function getEntryType($importer, $item)
 			 */
 			if ($is_a_remote_action && $community && (!$parent["forum_mode"]) && (!$parent["wall"])) {
 				$is_a_remote_action = false;
-				Logger::log("not a community action");
+				Logger::notice("not a community action");
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php:1531]:
```suggestion
				Logger::info("not a community action");
```

File: src/Protocol/DFRN.php [hunk 13]
```diff
@@ -1618,22 +1618,22 @@ private static function processVerbs($entrytype, $importer, &$item, &$is_like)
 			// Big question: Do we need these functions? They were part of the "consume_feed" function.
 			// This function once was responsible for DFRN and OStatus.
 			if ($activity->match($item["verb"], Activity::FOLLOW)) {
-				Logger::log("New follower");
+				Logger::notice("New follower");
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php:1624]:
```suggestion
				Logger::info("New follower");
```

File: src/Protocol/DFRN.php [hunk 14]
```diff
@@ -1618,22 +1618,22 @@ private static function processVerbs($entrytype, $importer, &$item, &$is_like)
...
 				Contact::addRelationship($importer, $contact, $item);
 				return false;
 			}
 			if ($activity->match($item["verb"], Activity::UNFOLLOW)) {
-				Logger::log("Lost follower");
+				Logger::notice("Lost follower");
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php:1629]:
```suggestion
				Logger::info("Lost follower");
```

File: src/Protocol/DFRN.php [hunk 15]
```diff
@@ -1618,22 +1618,22 @@ private static function processVerbs($entrytype, $importer, &$item, &$is_like)
...
 				Contact::removeFollower($contact);
 				return false;
 			}
 			if ($activity->match($item["verb"], Activity::REQ_FRIEND)) {
-				Logger::log("New friend request");
+				Logger::notice("New friend request");
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php:1634]:
```suggestion
				Logger::info("New friend request");
```

File: src/Protocol/DFRN.php [hunk 16]
```diff
@@ -1618,22 +1618,22 @@ private static function processVerbs($entrytype, $importer, &$item, &$is_like)
...
 				Contact::addRelationship($importer, $contact, $item, true);
 				return false;
 			}
 			if ($activity->match($item["verb"], Activity::UNFRIEND)) {
-				Logger::log("Lost sharer");
+				Logger::notice("Lost sharer");
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php:1639]:
```suggestion
				Logger::info("Lost sharer");
```

File: src/Protocol/DFRN.php [hunk 17]
```diff
@@ -1765,7 +1765,7 @@ private static function isSolicitedMessage(array $item)
 	 */
 	private static function processEntry($header, $xpath, $entry, $importer, $xml, $protocol)
 	{
-		Logger::log("Processing entries");
+		Logger::notice("Processing entries");
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php:1771]:
```suggestion
		Logger::info("Processing entries");
```

File: src/Protocol/DFRN.php [hunk 18]
```diff
@@ -2118,7 +2118,7 @@ private static function processEntry($header, $xpath, $entry, $importer, $xml, $
 	 */
 	private static function processDeletion($xpath, $deletion, $importer)
 	{
-		Logger::log("Processing deletions");
+		Logger::notice("Processing deletions");
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php:2124]:
```suggestion
		Logger::info("Processing deletions");
```

File: src/Protocol/DFRN.php [hunk 19]
```diff
@@ -2357,7 +2357,7 @@ private static function tgroupCheck($uid, $item)
 			foreach ($matches as $mtch) {
 				if (Strings::compareLink($link, $mtch[1]) || Strings::compareLink($dlink, $mtch[1])) {
 					$mention = true;
-					Logger::log('mention found: ' . $mtch[2]);
+					Logger::notice('mention found: ' . $mtch[2]);
```
Comment (MrPetovan) [inline src/Protocol/DFRN.php:2363]:
```suggestion
					Logger::info('mention found: ' . $mtch[2]);
```

File: src/Protocol/OStatus.php [hunk 20]
```diff
@@ -535,18 +535,18 @@ private static function process($xml, array $importer, array &$contact = null, &
...
 						if ($found) {
-							Logger::log("Item with uri ".$item["uri"]." for user ".$importer["uid"]." already exists.", Logger::DEBUG);
+							Logger::info("Item with uri ".$item["uri"]." for user ".$importer["uid"]." already exists.");
 						} elseif ($item['contact-id'] < 0) {
-							Logger::log("Item with uri ".$item["uri"]." is from a blocked contact.", Logger::DEBUG);
+							Logger::info("Item with uri ".$item["uri"]." is from a blocked contact.");
 						} else {
 							$ret = Item::insert($item);
-							Logger::log("Item with uri ".$item["uri"]." for user ".$importer["uid"].' stored. Return value: '.$ret);
+							Logger::notice("Item with uri ".$item["uri"]." for user ".$importer["uid"].' stored. Return value: '.$ret);
```
Comment (MrPetovan) [inline src/Protocol/OStatus.php]:
```suggestion
							Logger::info("Item with uri ".$item["uri"]." for user ".$importer["uid"].' stored. Return value: '.$ret);
```

File: src/Protocol/OStatus.php [hunk 21]
```diff
@@ -535,18 +535,18 @@ private static function process($xml, array $importer, array &$contact = null, &
 					foreach (self::$itemlist as $item) {
 						$found = Post::exists(['uid' => $importer["uid"], 'uri' => $item["uri"]]);
 						if ($found) {
-							Logger::log("Item with uri ".$item["uri"]." for user ".$importer["uid"]." already exists.", Logger::DEBUG);
+							Logger::info("Item with uri ".$item["uri"]." for user ".$importer["uid"]." already exists.");
 						} elseif ($item['contact-id'] < 0) {
-							Logger::log("Item with uri ".$item["uri"]." is from a blocked contact.", Logger::DEBUG);
+							Logger::info("Item with uri ".$item["uri"]." is from a blocked contact.");
```
Comment (MrPetovan) [inline src/Protocol/OStatus.php]:
```suggestion
							Logger::notice("Item with uri ".$item["uri"]." is from a blocked contact.");
```

File: src/Protocol/OStatus.php [hunk 22]
```diff
@@ -535,18 +535,18 @@ private static function process($xml, array $importer, array &$contact = null, &
 					foreach (self::$itemlist as $item) {
 						$found = Post::exists(['uid' => $importer["uid"], 'uri' => $item["uri"]]);
 						if ($found) {
-							Logger::log("Item with uri ".$item["uri"]." for user ".$importer["uid"]." already exists.", Logger::DEBUG);
+							Logger::info("Item with uri ".$item["uri"]." for user ".$importer["uid"]." already exists.");
```
Comment (MrPetovan) [inline src/Protocol/OStatus.php]:
```suggestion
							Logger::notice("Item with uri ".$item["uri"]." for user ".$importer["uid"]." already exists.");
```

File: src/Protocol/Salmon.php [hunk 23]
```diff
@@ -46,7 +46,7 @@ public static function getKey($uri, $keyhash)
 	{
 		$ret = [];
 
-		Logger::log('Fetching salmon key for '.$uri);
+		Logger::notice('Fetching salmon key for '.$uri);
```
Comment (MrPetovan) [inline src/Protocol/Salmon.php]:
```suggestion
		Logger::info('Fetching salmon key for '.$uri);
```

File: src/Protocol/Salmon.php [hunk 24]
```diff
@@ -116,12 +116,12 @@ public static function slapper($owner, $url, $slap)
...
 
 		if (!$owner['sprvkey']) {
-			Logger::log(sprintf("user '%s' (%d) does not have a salmon private key. Send failed.",
+			Logger::notice(sprintf("user '%s' (%d) does not have a salmon private key. Send failed.",
 			$owner['name'], $owner['uid']));
 			return;
 		}
 
-		Logger::log('slapper called for '.$url.'. Data: ' . $slap);
+		Logger::notice('slapper called for '.$url.'. Data: ' . $slap);
```
Comment (MrPetovan) [inline src/Protocol/Salmon.php]:
```suggestion
		Logger::info('slapper called for '.$url.'. Data: ' . $slap);
```

File: src/Protocol/Salmon.php [hunk 25]
```diff
@@ -209,7 +209,7 @@ public static function slapper($owner, $url, $slap)
 			$return_code = $postResult->getReturnCode();
 		}
 
-		Logger::log('slapper for '.$url.' returned ' . $return_code);
+		Logger::notice('slapper for '.$url.' returned ' . $return_code);
```
Comment (MrPetovan) [inline src/Protocol/Salmon.php]:
```suggestion
		Logger::info('slapper for '.$url.' returned ' . $return_code);
```

File: src/Util/DateTimeFormat.php [hunk 26]
```diff
@@ -157,7 +157,7 @@ public static function convert($s = 'now', $tz_to = 'UTC', $tz_from = 'UTC', $fo
 		try {
 			$d = new DateTime($s, $from_obj);
 		} catch (Exception $e) {
-			Logger::log('DateTimeFormat::convert: exception: ' . $e->getMessage());
+			Logger::notice('DateTimeFormat::convert: exception: ' . $e->getMessage());
```
Comment (MrPetovan) [inline src/Util/DateTimeFormat.php:160]:
```suggestion
			Logger::warning('DateTimeFormat::convert: exception: ' . $e->getMessage());
```

File: src/Util/XML.php [hunk 27]
```diff
@@ -268,7 +268,7 @@ public static function toArray($contents, $namespaces = true, $get_attributes =
 		}
 
 		if (!function_exists('xml_parser_create')) {
-			Logger::log('Xml::toArray: parser function missing');
+			Logger::notice('Xml::toArray: parser function missing');
```
Comment (MrPetovan) [inline src/Util/XML.php:271]:
```suggestion
			Logger::critical('Xml::toArray: parser function missing');
```

File: src/Util/XML.php [hunk 28]
```diff
@@ -283,7 +283,7 @@ public static function toArray($contents, $namespaces = true, $get_attributes =
 		}
 
 		if (! $parser) {
-			Logger::log('Xml::toArray: xml_parser_create: no resource');
+			Logger::notice('Xml::toArray: xml_parser_create: no resource');
```
Comment (MrPetovan) [inline src/Util/XML.php:286]:
```suggestion
			Logger::warning('Xml::toArray: xml_parser_create: no resource');
```

File: src/Worker/CheckVersion.php [hunk 29]
```diff
@@ -35,7 +35,7 @@ class CheckVersion
 {
 	public static function execute()
 	{
-		Logger::log('checkversion: start');
+		Logger::notice('checkversion: start');
```
Comment (MrPetovan) [inline src/Worker/CheckVersion.php:38]:
```suggestion
		Logger::info('checkversion: start');
```

File: src/Worker/CheckVersion.php [hunk 30]
```diff
@@ -51,15 +51,15 @@ public static function execute()
 				// don't check
 				return;
 		}
-		Logger::log("Checking VERSION from: ".$checked_url, Logger::DEBUG);
+		Logger::info("Checking VERSION from: ".$checked_url);
 
 		// fetch the VERSION file
 		$gitversion = DBA::escape(trim(DI::httpClient()->fetch($checked_url)));
-		Logger::log("Upstream VERSION is: ".$gitversion, Logger::DEBUG);
+		Logger::notice("Upstream VERSION is: ".$gitversion);
```
Comment (MrPetovan) [inline src/Worker/CheckVersion.php:58]:
```suggestion
		Logger::info("Upstream VERSION is: ".$gitversion);
```

File: src/Worker/CheckVersion.php [hunk 31]
```diff
@@ -51,15 +51,15 @@ public static function execute()
...
 
 		DI::config()->set('system', 'git_friendica_version', $gitversion);
 
-		Logger::log('checkversion: end');
+		Logger::notice('checkversion: end');
```
Comment (MrPetovan) [inline src/Worker/CheckVersion.php:62]:
```suggestion
		Logger::info('checkversion: end');
```

File: src/Worker/PubSubPublish.php [hunk 32]
```diff
@@ -66,17 +66,17 @@ private static function publish($id)
...
 
-		Logger::log('POST ' . print_r($headers, true) . "\n" . $params, Logger::DATA);
+		Logger::debug('POST ' . ['headers' => $headers, 'params' => $params]);
 
 		$postResult = DI::httpClient()->post($subscriber['callback_url'], $params, $headers);
 		$ret = $postResult->getReturnCode();
 
 		if ($ret >= 200 && $ret <= 299) {
-			Logger::log('Successfully pushed to ' . $subscriber['callback_url']);
+			Logger::notice('Successfully pushed to ' . $subscriber['callback_url']);
```
Comment (MrPetovan) [inline src/Worker/PubSubPublish.php]:
```suggestion
			Logger::info('Successfully pushed to ' . $subscriber['callback_url']);
```

File: view/theme/frio/theme.php [hunk 33]
```diff
@@ -57,7 +57,7 @@ function frio_install()
 	Hook::register('nav_info', 'view/theme/frio/theme.php', 'frio_remote_nav');
 	Hook::register('display_item', 'view/theme/frio/theme.php', 'frio_display_item');
 
-	Logger::log('installed theme frio');
+	Logger::notice('installed theme frio');
```
Comment (MrPetovan) [inline view/theme/frio/theme.php]:
```suggestion
	Logger::info('installed theme frio');
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]
I've suggested some logger severity changes. Here are my guidelines:
- `info`: Expected successful code path
- `notice`: Expected unsuccessful code path
- `warning`: Unexpected unsuccessful code path
- `critical`: Crippling Friendica capabilities

**Comment by nupplaphil** [comment]
Didn't you miss error, do you?

There's a official guideline with examples here: https://github.com/php-fig/log/blob/master/src/LoggerInterface.php

This is how I set the levels

**Comment by MrPetovan** [comment]
I did miss the `error` level indeed but the way it's worded sounds like the PHP Notice and Warnings that are currently logged separately.

**Comment by annando** [comment]
I decide between `info` and `notice` mostly by volume. Means: for a process like the receiving of a message I want to have a full trace of processes with `info`. But with `notice` I want have some bread crumbs. Means I want to at least have some information that a message arrived. One should say that with `notice` I want to see *if* the system is working and with `info`I want to see *how* it is working.

**Comment by MrPetovan** [comment]
This is pretty much what I said with `info` and `debug` for the detailed how. The simply guidelines I suggested were inspired by the existing examples, it just isn't applied consistently.

**Comment by annando** [comment]
When we are reconsidering where to use which log levels, we shouldn't do this only for this few legacy `log` calls but for all the `notice` calls that are out there. Also we then had to consider move several `info` calls to `debug`.

**Comment by MrPetovan** [comment]
So you won't accept my suggestions in this PR?

**Comment by annando** [comment]
I don't know. One the one hand I see that we have to have a deep look at what messages we expect at which log levels. On the other hand moving several log messages from "notice" to "info" can lead in losing important log messages for regular servers that currently will most likely run with "notice" because "info" is too crowded. (But on the other hand I don't want to make your contribution to this PR obsolete)

**Comment by MrPetovan** [comment]
Thank you for your consideration! What are the important messages that I downgraded from `notice` to `info`? I didn't change that many calls, it should be easy to spot the important calls that need to stay at the `notice` level.

**Comment by annando** [comment]
I will have a look at the suggestions, I guess.

File: src/Protocol/DFRN.php (hunk 8 above)
Comment (annando) [reply src/Protocol/DFRN.php:1298]: I would like to keep that. It is important for tracing incoming messages.

File: src/Protocol/DFRN.php (hunk 9 above)
Comment (annando) [reply src/Protocol/DFRN.php:1403]: Like above. It is important for tracing incoming messages.

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**